airflow/jobs.py

```diff
@@ -721,6 +721,25 @@ def create_dag_run(self, dag, session=None):
             if dag.schedule_interval == '@once' and last_scheduled_run:
                 return None
 
+            # don't do backfill for dag's that don't have dag.backfill = True
+            if not dag.backfill:
+                # The logic is that we move start_date up until
+                # one period before, so that datetime.now() is AFTER
+                # the period end, and the job can be created...
```

Comment:
I guess this works, but it seems wrong to modify the `dag` argument (which was previously being used "read-only"). I think `create_dag_run` would be cleaner if we refactored it to avoid having to do this.